### ISSUE TITLE
Paid centers always show booked, but slots never open

### ISSUE THREAD

**Issue by muniamma** [cowinapi/developer.cowin#54]
Paid centers like Jayanagar Apollo hospitals alway show booked against any date. But, I have never seen them open, even though for the past one week, every day, I check at frequent intervals from 430am to 11pm IST. Are hackers booking in the backend?

**Comment by monish-1234** [comment]
That's not easily possible. It's most probably done by general people. As far as I've seen, slots get booked up with 5 mins of opening. 

P.S if you're checking slots without logging in , you might expect to never see a open slot anytime soon as the Public API is cached for 30 mins. Login and check, you might luck out

**Comment by muniamma** [comment]
Thanks for the response. There are several messages floating around that
hacker  techies are writing python scripts to find out open slots and book
them. There are  many people (including businesses) who are accessing your
APIs to find out open slots. You might want to check it out (from logs or
other means).

Regards,

G Padmanabhan

**Comment by monish-1234** [comment]
Unfortunately that resides with the developers @cowinapi.
Personally I would say it's okay that people are trying to get other's notified as soon as slots open up but the booking system should reside only with Cowin. 

TLDR. If the devs reduce the cache time for the Public, You can get live data without logging in 

Anyway, If I answered your query , help the devs by closing the issue.

**Comment by muniamma** [comment]
Notification is ok. Not booking. I am a user. Developers should check this
out.

G Padmanabhan

**Comment by muniamma** [comment]
I am a user. Let developers answer this.

G Padmanabhan

**Comment by chandraaditya** [comment]
That is not true, I have seen paid slots open up and even get booked. And your information about techies and businesses accessing APIs and booking is false too. https://twitter.com/umanghome/status/1390892481911279616 if you look at that thread, you can clearly see most of the slots are being booked by verified humans

**Comment by monish-1234** [comment]
> Notification is ok. Not booking. I am a user. Developers should check this out. G Padmanabhan

It is already done. The Devs implemented captacha. Which would make the bots obsolete , even the one's that solve the captacha.

As you say you're just a user and not a dev, I'll explain it in simple terms.

What a bot does is constantly check for open slots (probably every 2-5 mins) using the Slot Availability API . Which itself is cached for 30 mins. Meaning,  when a slot opens inside the website (after logging in) , it'll take about 30 mins to update it in the public API which everyone uses to notify.

So, Now people and the bots get notified only after 30 mins. What a bot then does is ask your phone number and call the OTP API to send you a SMS. After you enter the OTP (this is where things get interesting) the USED TO book the slot within seconds. 

But, the devs have now implemented Captcha to prevent that. Bots can't solve captcha (easily). So what they do is give the captcha to the user after which the user should solve them then it proceeds to booking .
Thus increasing the time for booking. So manually booking and booking via bots would yield a maximum of 5-10 second difference.

I don't think the website, the devs, the people who notify and the people who automate booking is at fault here.
The issue lies in with the low availability of vaccines. Opening 100-200 slots in a city with 50 lakh people between 18-45 is the issue.

**Comment by muniamma** [comment]
Looks like this was the issue. Good move by govt.
https://www.indiatoday.in/technology/news/story/changes-in-cowin-app-govt-restricts-vaccine-slot-info-to-fight-bots-and-alert-services-1799827-2021-05-07

G Padmanabhan
